### PR TITLE
DAOS-8916 client: Fix potential OFI interface/domain mismatch (#7183)

### PR DESCRIPTION
**Pull request by kjacque**
The network interface and domain must be acquired from the same
source, either the agent automatic config, or the OFI_INTERFACE
and OFI_DOMAIN environment variables.

Replace the use of OFI_INTERFACE to provide a default fabric_iface value
to the server config with a new DAOS_TEST_FABRIC_IFACE environment
variable.  Launch.py will still use any existing OFI_INTERFACE value but
will no longer set the env.  In CI this will allow the launch node to
run without OFI_INTERFACE beign set.

Co-authored-by: Phillip Henderson <phillip.henderson@intel.com>
Signed-off-by: Kris Jacque <kristin.jacque@intel.com>

### CODE REVIEW

**Comment by kjacque** [comment]
Had to resolve a conflict in launch.py while backporting. @phender can you have a look, please, to make sure I didn't make any mistakes in that merge?

**Review by phender — CHANGES_REQUESTED** [review]
This commit includes part of the changes in https://github.com/daos-stack/daos/pull/7111, which is currently blocked by https://daosio.atlassian.net/browse/DAOS-8933.  Should we wait for 7111 before landing these changes as this was the order things were applied on the master branch?

**Pull request merged**
